protocols/kad/src/behaviour/test.rs

```diff
@@ -425,6 +425,68 @@ fn unresponsive_not_returned_indirect() {
     }))
 }
 
+#[test]
+// Test the result of get_closest_peers with different num_results
+// Note that the result is capped after exceeds K_VALUE
+fn get_closest_with_different_num_results() {
+    let k_value = K_VALUE.get();
+    for replication_factor in [5, k_value / 2, k_value] {
+        for num_results in k_value / 2..k_value * 2 {
+            get_closest_with_different_num_results_inner(num_results, replication_factor)
```

Comment:
this function is not used elsewhere right? We can inline its code here

Reply:
make it a separate function to avoid the code becomes too `nested`
it's for a better reading experience and coding format.

I'd prefer not to inline the code block here, unless really `have to` ?